packages/block-editor/src/hooks/gap.js

```diff
@@ -38,6 +39,45 @@ export function hasGapValue( props ) {
 	return props.attributes.style?.spacing?.blockGap !== undefined;
 }
 
+/**
+ * Returns a BoxControl object value from a given blockGap style.
+ * The string check is for backwards compatibility before Gutenberg supported
+ * split gap values (row and column) and the value was a string n + unit.
+ *
+ * @param {string? | Object?} rawBlockGapValue A style object.
+ * @return {Object?}                           A value to pass to the BoxControl component.
+ */
+export function getGapValueFromStyle( rawBlockGapValue ) {
+	if ( ! rawBlockGapValue ) {
+		return rawBlockGapValue;
+	}
+
+	const isValueString = typeof rawBlockGapValue === 'string';
+	return cleanEmptyObject( {
+		top: isValueString ? rawBlockGapValue : rawBlockGapValue?.top,
+		left: isValueString ? rawBlockGapValue : rawBlockGapValue?.left,
+	} );
+}
+
+/**
+ * Returns a CSS value for the `gap` property from a given blockGap style.
+ *
+ * @param {string? | Object?} blockGapValue A style object.
+ * @param {string?}           defaultValue  A default gap value.
+ * @return {string?}                        The concatenated gap value (row and column).
+ */
+export function getGapCSSValue( blockGapValue, defaultValue = '0' ) {
+	const blockGapBoxControlValue = getGapValueFromStyle( blockGapValue );
+	if ( ! blockGapBoxControlValue ) {
+		return blockGapBoxControlValue;
+	}
+
+	const row = blockGapBoxControlValue?.top || defaultValue;
+	const column = blockGapBoxControlValue?.left || defaultValue;
+
+	return row === column ? row : `${ row } ${ column }`;
+}
```

Comment:
Here I was wondering whether we should be returning the full definition so that we can isolate row and column values where only one is set, e.g., 

```js
// row and column is set or is the same
return `gap: 10px 10px;`;

// only row is set
return  `row-gap: 10px;`;

// only column is set
return  `column-gap: 10px;`;
```

That way, if a block ever has to inherit either row/column we're not overwriting it. 

Just typing out loud...

Reply:
After some further tinkering I probably need to put more thought into this.

I'm picturing a future where theme.json authors could add something like this.

```json
"spacing": {
    "blockGap": {
         "row": "0.5em",
         "column": "2em"
     }
}
```

and blocks would then inherit or use these values as defaults. 

Currently "blockGap" accept a string, which can be really anything, including the shorthand value of `"0.5em 2em"` so it might take some more cogitating to make things backwards compatible.

Reply:
Sure, we don't need it for this PR; better to work out the details as a future enhancement.